java/code/src/com/suse/manager/virtualization/GuestDefinition.java

```diff
@@ -0,0 +1,254 @@
+     *
+     * Don't forget to change the maximum before raising the current amount
+     * of memory or it will be capped.
+     *
+     * @param memoryIn current amount of memory
+     */
+    public void setMemory(long memoryIn) {
+        this.memory = Math.min(this.maxMemory, memoryIn);
+    }
+
+    /**
+     * @return the maximum amount of memory allocated to the guest in KiB
+     */
+    public long getMaxMemory() {
+        return maxMemory;
+    }
+
+    /**
+     * Set the maximum number of memory in KiB.
+     *
+     * If setting a value smaller than the current amount of memory, the
+     * latter will be adjusted.
+     *
+     * @param maxIn maximum number of Virtual CPUs
+     */
+    public void setMaxMemory(long maxIn) {
+        this.maxMemory = maxIn;
+        this.memory = Math.min(this.maxMemory, this.memory);
+    }
+
+    /**
+     * @return Vcpu definition
+     */
+    public GuestVcpuDef getVcpu() {
+        return vcpu;
+    }
+
+    /**
+     * @param vcpuIn The vcpu to set.
+     */
+    public void setVcpu(GuestVcpuDef vcpuIn) {
+        vcpu = vcpuIn;
+    }
+
+    /**
+     * @return OS definition
+     */
+    public GuestOsDef getOs() {
+        return os;
+    }
+
+    /**
+     * @param osIn The os to set.
+     */
+    public void setOs(GuestOsDef osIn) {
+        os = osIn;
+    }
+
+    /**
+     * Compute the virtual instance type from the VM OS definition.
+     *
+     * @return the VirtualInstanceType
+     */
+    public VirtualInstanceType getVirtualInstanceType() {
+        // FIXME This logic is actually not perfect, but matches the legacy one
+        VirtualInstanceType virtType = VirtualInstanceFactory.getInstance().getFullyVirtType();
+        if (!getOs().getType().equals("hvm")) {
+            virtType = VirtualInstanceFactory.getInstance().getParaVirtType();
+        }
+        return virtType;
+    }
+
+    /**
+     * Create a new VM definition from a libvirt XML description.
+     *
+     * @param xmlDef libvirt XML domain definition
+     * @return parsed definition or {@code null}
+     */
+    public static GuestDefinition parse(String xmlDef) {
+        GuestDefinition def = null;
+        SAXBuilder builder = new SAXBuilder();
+        builder.setValidation(false);
+
+        try {
+            Document doc = builder.build(new StringReader(xmlDef));
+            def = new GuestDefinition();
+
+            Element domainElement = doc.getRootElement();
+            def.type = domainElement.getAttributeValue("type");
+            def.setName(domainElement.getChildText("name"));
+            def.uuid = domainElement.getChildText("uuid");
+            def.setMemory(parseMemory(domainElement.getChild("currentMemory")));
+            def.setMaxMemory(parseMemory(domainElement.getChild("memory")));
+
+            def.vcpu = GuestVcpuDef.parse(domainElement.getChild("vcpu"));
+            def.os = GuestOsDef.parse(domainElement.getChild("os"));
+        }
+        catch (JDOMException e) {
```

Comment:
Shouldn't we log the error in case we catch these `Exception`s?

Reply:
Add a trace in the log